sru/modules.py

```diff
@@ -361,22 +210,95 @@ def forward(self, input, c0=None, mask_pad=None, **kwargs):
 
         # apply layer norm before activation (i.e. before SRU computation)
         residual = input
-        if self.layer_norm:
+        if self.layer_norm is not None:
             input = self.layer_norm(input)
 
         # apply dropout for multiplication
         if self.training and (self.rnn_dropout > 0):
             mask = self.get_dropout_mask_((batch_size, input.size(-1)), self.rnn_dropout)
             input = input * mask.expand_as(input)
 
+        # get the scaling constant; scale_x is a scalar
+        scale_val = self.scale_x if self.rescale else None
+
+        # get dropout mask
+        if self.training and (self.dropout > 0):
+            mask_c = self.get_dropout_mask_((batch_size, self.output_size), self.dropout)
+        else:
+            mask_c = None
+
         # compute U, V
         #   U is (length, batch_size, output_size * num_matrices)
         #   V is (output_size*2,) or (length, batch_size, output_size * 2) if provided
+        U, V = self.compute_UV(input, c0, mask_pad)
+
+        # apply elementwise recurrence to get hidden states h and c
+        h, c = self.apply_recurrence(U, V,
+                                     residual, c0,
+                                     scale_val,
+                                     mask_c,
+                                     mask_pad)
+        return h, c
+
+    def apply_recurrence(self,
+                         U: Tensor,
```

Comment:
nit: having capitalized variables makes me think they're cosntants. Possibly worth lowercasing?  (though I do see the use in this case, so up to you)